src/library/scala/collection/mutable/HashTable.scala

```diff
@@ -365,9 +365,8 @@ trait HashTable[A, Entry >: Null <: HashEntry[A, Entry]] extends HashTable.HashU
   // this is of crucial importance when populating the table in parallel
   protected final def index(hcode: Int) = {
     val ones = table.length - 1
-    val improved = improve(hcode, seedvalue)
-    val shifted = (improved >> (32 - java.lang.Integer.bitCount(ones))) & ones
-    shifted
+    val exponent = Integer.numberOfLeadingZeros(ones)
```

Comment:
Speed improvement or no, when the goal is to count the number of leading zeros it's better to clean up the method and say `numberOfLeadingZeros` unless there's a significant performance hit.  It's nice that it's faster too, though!